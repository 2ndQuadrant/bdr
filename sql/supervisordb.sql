\c postgres

-- The DB name bdr_supervisordb is reserved by BDR. None
-- of these commands may be permitted.

CREATE DATABASE bdr_supervisordb;

DROP DATABASE bdr_supervisordb;

ALTER DATABASE bdr_supervisordb RENAME TO someothername;

ALTER DATABASE regression RENAME TO bdr_supervisordb;

-- We can connect to the supervisor db...
\c bdr_supervisordb

SET log_statement = 'all';

-- We actually did connect
SELECT current_database();

-- And do read-only work
SELECT 1;

-- but not do anything interesting
CREATE TABLE create_fails(id integer);

\d

-- except vacuum
VACUUM;
