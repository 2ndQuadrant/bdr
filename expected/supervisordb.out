\c postgres
-- The DB name bdr_supervisordb is reserved by BDR. None
-- of these commands may be permitted.
CREATE DATABASE bdr_supervisordb;
ERROR:  The BDR extension reserves the database name bdr_supervisordb for its own use
HINT:  Use a different database name
DROP DATABASE bdr_supervisordb;
ERROR:  The BDR extension reserves the database name bdr_supervisordb for its own use
HINT:  Use a different database name
ALTER DATABASE bdr_supervisordb RENAME TO someothername;
ERROR:  The BDR extension reserves the database name bdr_supervisordb for its own use
HINT:  Use a different database name
ALTER DATABASE regression RENAME TO bdr_supervisordb;
ERROR:  The BDR extension reserves the database name bdr_supervisordb for its own use
HINT:  Use a different database name
-- We can connect to the supervisor db...
\c bdr_supervisordb
SET log_statement = 'all';
ERROR:  No commands may be run on the BDR supervisor database
-- We actually did connect
SELECT current_database();
 current_database 
------------------
 bdr_supervisordb
(1 row)

-- And do read-only work
SELECT 1;
 ?column? 
----------
        1
(1 row)

-- but not do anything interesting
CREATE TABLE create_fails(id integer);
ERROR:  No commands may be run on the BDR supervisor database
\d
      List of relations
 Schema | Name | Type | Owner 
--------+------+------+-------
(0 rows)

-- except vacuum
VACUUM;
